### ISSUE TITLE
Graceful algorithm degradation

### ISSUE THREAD

**Issue by happy-turtle** [happy-turtle/oit-unity#12]
The current implementation and graphics pipeline structure lead to a problem: A fixed per-pixel list length has to be set before rendering. That means if during render too many fragments overlap at one pixel location the algorithm will fail at correctly blending those pixels. 
<img src="https://user-images.githubusercontent.com/18415215/197854471-a494d102-556a-4b25-ae25-257f22d0dce2.png" width="40%" height="40%">
There is no way to dynamically set the pixel list length. The alternative, which is already proposed and implemented by others, is to gracefully degrade the transparency blending. Visual coherency is retained at the cost of losing alpha blending accuracy. I am not sure if this requires additional synchronization, some papers mention requiring pixel synchronization for this approach.

To solve this problem we would need to find an algorithm that can be implemented with Unity. Some research suggestions:
- [AOIT-Update](https://github.com/GameTechDev/AOIT-Update)
- [Multi-Layer Alpha Blending](https://www.intel.com/content/www/us/en/developer/articles/technical/multi-layer-alpha-blending.html)

**Comment by happy-turtle** [comment]
> maybe to make a min stack to store closest pixels in linked list and drop the far pixels is a good way.

Good idea! I think a problem might be how we can detect which pixel is further away? We currently only write to the list at list creation, but comparing the pixel depth will require _reading_ the list at the same time. And then you can't be sure if all other pixels are already in the list? This would probably require some kind of atomic operation.

**Comment by SnowWindSaveYou** [comment]
> > maybe to make a min stack to store closest pixels in linked list and drop the far pixels is a good way.
> 
> Good idea! I think a problem might be how we can detect which pixel is further away? We currently only write to the list at list creation, but comparing the pixel depth will require _reading_ the list at the same time. And then you can't be sure if all other pixels are already in the list? This would probably require some kind of atomic operation.

the limitation of the list size only happens when store linked list result to a block list, the  linked list self doesn't have this problem.  
and the min stack is a way to filter smallest values when reading list, we can store any length of linked list and filter them into a fixed size list, it's a simple and efficient algorithm on cpu but i'm not sure it's performance on gpu... maybe try it after i submit my homework...
